Assets/Scripts/Waypoint Scripts/UpdateWaypoints.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+namespace ISAACS
+{
+    public class UpdateWaypoints : MonoBehaviour
+    {
+
+        public Material selectedPassedLine;
+        public Material unselectedPassedLine;
+
+        public GameObject targetWaypoint;
+        public int currentWaypoint;
+        public GameObject prevPoint;
+
+        private int totalWaypoints;
+        private LineRenderer line;
+        private Drone thisDrone;
+
+        void Start()
+        {
+            thisDrone = gameObject.GetComponent<DroneProperties>().droneClassPointer;
+            line = this.GetComponentInParent<LineRenderer>();
+        }
+
+        void Update()
+        {
+            totalWaypoints = thisDrone.waypoints.Count - 1;
+
+            if (prevPoint != null)
+            {
+                DisplayPastPath();
+            }
+
+            SelectTarget();
+        }
+
+        private void DisplayPastPath()
+        {
+            line.SetPosition(0, this.transform.position);
+
+            if (thisDrone.waypoints.Count == 0)
+            {
+                line.SetPosition(1, this.transform.position);
+            }
+            else
+            {
+                line.SetPosition(1, prevPoint.transform.position);
+            }
+
+            if (thisDrone.selected)
+            {
+                line.material = selectedPassedLine;
+            }
+            else
+            {
+                line.material = unselectedPassedLine;
+            }
+        }
+
+        private void SelectTarget()
```

Comment:
Please add comments for each function

Reply:
We will work on the waypoint scripts in a new pull request.